melkor/models/sklearn_pipeline.py

```diff
@@ -0,0 +1,110 @@
+import sklearn
+import pandas as pd
+import numpy as np
+import pickle
+from nptyping import NDArray, Float32
+from typing import Union
+from sklearn.pipeline import Pipeline
+from sklearn.compose import ColumnTransformer
+
+# TODO: Add scoring metrics
+# TODO: Change y to 1d array
+
+
+class SKLearnModelPipeline:
+    def __init__(
+        self,
+        config: dict = {},
+        cat_cols: list = [],
+        num_cols: list = [],
+        inference: bool = False,
+    ) -> None:
+        self.config = config
+        self.cat_cols = cat_cols
+        self.num_cols = num_cols
+        if not inference:
+            self.init_train()
+
+    def init_train(self):
+        self.config = self.config.get("model_pipeline")
+        self.pipeline = self.create_pipeline()
+
+    def create_pipeline(self):
+
+        cat_transforms = self.config.get("data_transformation").get("categorical")
+        num_transforms = self.config.get("data_transformation").get("numerical")
+
+        categorical_transformer = self.parse_subpipeline(cat_transforms)
+        numeric_transformer = self.parse_subpipeline(num_transforms)
+
+        data_transformation = ColumnTransformer(
+            transformers=[
+                ("num", numeric_transformer, self.num_cols),
+                ("cat", categorical_transformer, self.cat_cols),
+            ]
+        )
+
+        model_pipeline = self.parse_subpipeline({"model": self.config.get("model")})
+
+        full_pipeline = Pipeline(
```

Comment:
Why all the wrapping over, if there is no real extension to the underlying sklearn.Pipeline object?